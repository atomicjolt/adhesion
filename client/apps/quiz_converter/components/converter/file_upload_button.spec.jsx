import React from 'react';
import { shallow } from 'enzyme';
import FileUploadButton from './file_upload_button';

describe('File Upload Button', () => {
  let result;
  let called;

  beforeEach(() => {
    called = false;
    const props = {
      text: 'Button text',
      htmlId: 'quiz',
      selectFile: () => { called = true; },
    };

    result = shallow(<FileUploadButton {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('calls selectFile when a file is selected', () => {
    const input = result.find('input');
    input.simulate('change', { target: { files: ['file'] } });
    expect(called).toBeTruthy();
  });

  it('does not call selectFile when there are no files selected', () => {
    const input = result.find('input');
    input.simulate('change', { target: { files: [] } });
    expect(called).toBeFalsy();
  });
});
