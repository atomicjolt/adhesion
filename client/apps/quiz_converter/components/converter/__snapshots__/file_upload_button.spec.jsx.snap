// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Upload Button matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FileUploadButton
    htmlId="quiz"
    selectFile={[Function]}
    text="Button text"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="c-upload"
          id="quiz"
          name="quiz"
          onChange={[Function]}
          type="file"
/>,
        <label
          className="c-quiz-label"
          htmlFor="quiz"
>
          <span>
                    Button text
          </span>
</label>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "c-upload",
          "id": "quiz",
          "name": "quiz",
          "onChange": [Function],
          "type": "file",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            Button text
</span>,
          "className": "c-quiz-label",
          "htmlFor": "quiz",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Button text",
          },
          "ref": null,
          "rendered": "Button text",
          "type": "span",
        },
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="c-upload"
            id="quiz"
            name="quiz"
            onChange={[Function]}
            type="file"
/>,
          <label
            className="c-quiz-label"
            htmlFor="quiz"
>
            <span>
                        Button text
            </span>
</label>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "c-upload",
            "id": "quiz",
            "name": "quiz",
            "onChange": [Function],
            "type": "file",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              Button text
</span>,
            "className": "c-quiz-label",
            "htmlFor": "quiz",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Button text",
            },
            "ref": null,
            "rendered": "Button text",
            "type": "span",
          },
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
